osdep/macosx_application.m

```diff
@@ -362,6 +362,16 @@ int cocoa_main(int argc, char *argv[])
         ctx.argc     = &argc;
         ctx.argv     = &argv;
 
+        // convert all filenames and URLs to decomposed unicode (NFC->NFD).
+        // command line input can either be NFC or NFD whereas anything else
+        // on macOS uses NFD.
+        char *arg_decomposed;
+        for (int i = 1; i < argc; i++) {
+            NSString *arg = [NSString stringWithUTF8String:argv[i]];
+            arg_decomposed = [[arg decomposedStringWithCanonicalMapping] UTF8String];
```

Comment:
i did a bit of reading up since i am not too experienced with memory management.

[NSString stringWithUTF8String:argv[i]]
is apparently equivalent to
[[[NSString alloc] initWithUTF8String:argv[i]] autorelease]

since the whole main method is wrapped in an autoreleasepool, it will automatically free it at the end of that pool. the end itself is never reached according to the comment in the code in line 393. so this should be fine.